### PR TITLE
[release-1.5] 🌱 Bump go.opentelemetry.io/* dependencies

### PR DESCRIPTION
**Pull request by furkatgofurov7**
**What this PR does / why we need it**:
Trivy is failing https://github.com/kubernetes-sigs/cluster-api/actions/runs/6612918116/job/17961612952 still after we bumped golang.org/x/net to v0.17.0 in all branches. So bumping `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp` to v0.44.0 and running go mod tidy outputs:

```
sigs.k8s.io/cluster-api/test/infrastructure/inmemory/internal/server/api imports
        k8s.io/apiserver/pkg/server imports
        k8s.io/component-base/tracing imports
        go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc imports
        go.opentelemetry.io/otel/exporters/otlp/internal: module go.opentelemetry.io/otel/exporters/otlp@latest found (v0.20.1), but does not contain package go.opentelemetry.io/otel/exporters/otlp/internal
sigs.k8s.io/cluster-api/test/infrastructure/inmemory/internal/server/api imports
        k8s.io/apiserver/pkg/server imports
        k8s.io/component-base/tracing imports
        go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc imports
        go.opentelemetry.io/otel/exporters/otlp/otlptrace/internal/otlpconfig imports
        go.opentelemetry.io/otel/exporters/otlp/internal/envconfig: module go.opentelemetry.io/otel/exporters/otlp@latest found (v0.20.1), but does not contain package go.opentelemetry.io/otel/exporters/otlp/internal/envconfig
```

so I had to bump some other dependencies in go.opentelemetry.io/otel/* package to get pass the trivy scan workflow. Tested it in my fork and passed: https://github.com/furkatgofurov7/cluster-api/actions/runs/6613251793/job/17965058387. Hopefully it should pass after this change in our GH workflow

/area dependency

### CODE REVIEW

File: hack/tools/go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ replace sigs.k8s.io/cluster-api => ../../
 replace sigs.k8s.io/cluster-api/test => ../../test
 
 require (
-	cloud.google.com/go/storage v1.29.0
+	cloud.google.com/go/storage v1.30.1
```
Comment (furkatgofurov7) [inline hack/tools/go.mod:10]: Note: these and below deps were not bumped intentionally but automatically while running `go mod tidy` in the hack/tools folder

**Comment by sbueringer** [comment]
/lgtm
/approve

/hold
for additional coverage (not sure how we use the grpc)

/test pull-cluster-api-e2e-full-release-1-5

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

**Comment by furkatgofurov7** [comment]
@sbueringer e2e full is green, can we unhold it?

**Pull request merged**